### PR TITLE
Faster sync

### PR DESCRIPTION
**Pull request by karlb**
Improve event syncing speed by filtering more selectively. This brings down the initial sync on mainnet for a fresh node to 30s (parity, infura) to 2mins (geth).

## Before

Fetch all events for the relevant contracts and filter in the Raiden client

## After

* Fetch `TokenNetworkCreated` events from TN registry
* For each TN found, fetch the `ChannelOpened` events that include the node as participant
* For each of these channels, fetch the `ChannelClosed`, `ChannelSettled`, `ChannelNewDeposit` and `ChannelWithdraw` events
* Fetch all `SecretRevealed` events from the SecretRegistry once the node starts to participate in a channel

Closes https://github.com/raiden-network/raiden/issues/6497

The SP integration test must fail until https://github.com/raiden-network/scenario-player/pull/635 is merged, which has to be based on this PR. To break the cycle, I will force merge this when all tests but the SP integration are green.

Potential future optimizations if we need more speed:
* We could only query for secrets with a specific secrethash
* We could stop looking for new token networks one the registries limit is reached (we don't want to have limits in the long run, though)
* We could get some information form a thegraph.com subgraph
* We could look for new TNs in the background while we allow working on the found TNs
* We could save a list of tokens in the TN registry contract, so that we don't have to fetch the events

### CODE REVIEW

File: raiden/blockchain/filters.py [hunk 1]
```diff
@@ -86,3 +105,172 @@ def decode_event(abi: ABI, event_log: LogReceipt) -> EventData:
...
+            **{Address(tnr): tnr_abi for tnr in self.token_network_registry_addresses},
+            **{Address(tn): tn_abi for tn in self.token_network_addresses},
+        }
+        if self.secret_registry_address:
+            abis[Address(self.secret_registry_address)] = secret_registry_abi
+        return abis
+
+    def union(self, other: "RaidenContractFilter") -> "RaidenContractFilter":
+        """ Return a new RaidenContractFilter with all elements from both input filters """
+        # We must not have to different non-None secret registries. Choose the non-None one.
```
Comment (fredo) [inline raiden/blockchain/filters.py:249]:
```suggestion
        # We must not have two different non-None secret registries. Choose the non-None one.
```

File: raiden/transfer/state_change.py [hunk 2]
```diff
@@ -318,21 +318,6 @@ def token_network_address(self) -> TokenNetworkAddress:
 # FIXME: remove on next breaking release
```
Comment (fredo) [inline raiden/transfer/state_change.py:318]: Do we still need this comment?

**Review by fredo — APPROVED** [review]
LGTM, just a typo and a question left.

File: raiden/transfer/state_change.py (hunk 2 above)
Comment (palango) [reply raiden/transfer/state_change.py:318]: This (ContractReceiveRouteClosed) shouldn't be removed if the released is supposed to be backwards-compatible.

Comment (fredo) [reply raiden/transfer/state_change.py:318]: right, but it is removed in this PR isn't it?

Comment (palango) [reply raiden/transfer/state_change.py:318]: That's my point. Depending on what the goals are, this should eventually stay.

**Pull request merged**